driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1857,6 +1833,8 @@ public void onFailure(Throwable t) {
                     s.updateCreatedPools(blockingExecutor);
 
             } finally {
+                if (preExistentConnection != null && !preExistentConnection.hasPool())
```

Comment:
There are two things out of this try block: 1) checking the protocol version and 2) acquiring the notifications lock.

1) will never fail because we've already opened a connection to the node.
2) is an edge case (something is wrong in the driver's code), but we still want to properly close the connection in that case. If you don't want nested try blocks, you can do it in the if block.